src/mailmime_write_generic.rs

```diff
@@ -1146,14 +1107,11 @@ unsafe fn mailmime_part_write_driver(
                         16754986508692159943 => {}
                         _ => {
                             first = 1i32;
-                            cur = (*(*build_info).mm_data.mm_multipart.mm_mp_list).first;
-                            loop {
-                                if cur.is_null() {
-                                    current_block = 3546145585875536353;
-                                    break;
-                                }
-                                let mut subpart: *mut mailmime = 0 as *mut mailmime;
-                                subpart = (*cur).data as *mut mailmime;
+                            if (*(*build_info).mm_data.mm_multipart.mm_mp_list).is_empty() {
```

Comment:
I think this should be done unconditionally.

`current_block` is set to `3546145585875536353` if `cur` reaches end of the list successfully. It is set to `16754986508692159943` on error, it happens before every other `break`.

So the solution here is to set current_block to `3546145585875536353` before the loop and set it to `16754986508692159943` on error (that is already done). Basically, `3546145585875536353` is an `ok_to_continue == true` value, and `16754986508692159943` indicates an error.